Source/carthage/Build.swift

```diff
@@ -17,23 +17,25 @@ public struct BuildCommand: CommandType {
 	public struct Options: OptionsType {
 		public let configuration: String
 		public let buildPlatform: BuildPlatform
+		public let derivedDataPath: String
 		public let skipCurrent: Bool
 		public let colorOptions: ColorOptions
 		public let verbose: Bool
 		public let directoryPath: String
 		public let dependenciesToBuild: [String]?
 
-		public static func create(configuration: String) -> BuildPlatform -> Bool -> ColorOptions -> Bool -> String -> [String] -> Options {
-			return { buildPlatform in { skipCurrent in { colorOptions in { verbose in { directoryPath in { dependenciesToBuild in
+		public static func create(configuration: String) -> BuildPlatform -> String -> Bool -> ColorOptions -> Bool -> String -> [String] -> Options {
+			return { buildPlatform in { derivedDataPath in { skipCurrent in { colorOptions in { verbose in { directoryPath in { dependenciesToBuild in
 				let dependenciesToBuild: [String]? = dependenciesToBuild.isEmpty ? nil : dependenciesToBuild
-				return self.init(configuration: configuration, buildPlatform: buildPlatform, skipCurrent: skipCurrent, colorOptions: colorOptions, verbose: verbose, directoryPath: directoryPath, dependenciesToBuild: dependenciesToBuild)
-			} } } } } }
+				return self.init(configuration: configuration, buildPlatform: buildPlatform, derivedDataPath: derivedDataPath, skipCurrent: skipCurrent, colorOptions: colorOptions, verbose: verbose, directoryPath: directoryPath, dependenciesToBuild: dependenciesToBuild)
+			} } } } } } }
 		}
 
 		public static func evaluate(m: CommandMode) -> Result<Options, CommandantError<CarthageError>> {
 			return create
 				<*> m <| Option(key: "configuration", defaultValue: "Release", usage: "the Xcode configuration to build")
 				<*> m <| Option(key: "platform", defaultValue: .All, usage: "the platforms to build for (one of ‘all’, ‘Mac’, ‘iOS’, ‘watchOS’, 'tvOS', or comma-separated values of the formers except for ‘all’)")
+				<*> m <| Option(key: "derived-data", defaultValue: "", usage: "path to the custom derived data folder")
```

Comment:
When I use `nil` and `String` I receive an error:

```
Missing argument for --derived-data
```

According to the specification: 
> If this is nil, this option is always required.

Is there any workaround for this, or maybe I'm doing something wrong?